crates/services/tx_status_manager/src/manager.rs

```diff
@@ -0,0 +1,106 @@
+use anyhow::anyhow;
+use std::{
+    collections::HashMap,
+    sync::{
+        Arc,
+        Mutex,
+    },
+};
+
+use fuel_core_types::{
+    fuel_tx::{
+        Bytes32,
+        TxId,
+    },
+    services::txpool::TransactionStatus,
+};
+use tokio::sync::broadcast;
+
+use crate::{
+    error::Error,
+    tx_status_stream::{
+        TxStatusStream,
+        TxUpdate,
+    },
+    update_sender::{
+        MpscChannel,
+        TxStatusChange,
+    },
+};
+
+#[derive(Clone)]
+pub struct TxStatusManager {
+    statuses: Arc<Mutex<HashMap<TxId, TransactionStatus>>>,
+    tx_status_change: TxStatusChange,
+}
+
+impl TxStatusManager {
+    pub fn new(tx_status_change: TxStatusChange) -> Self {
+        Self {
+            statuses: Arc::new(Mutex::new(HashMap::new())),
+            tx_status_change,
+        }
+    }
+
+    pub fn status_update(&self, tx_id: TxId, tx_status: TransactionStatus) {
+        tracing::debug!(%tx_id, ?tx_status, "new tx status");
+
+        // TODO[RC]: Capacity checks? - Protected by TxPool capacity checks, except for the squeezed state. Maybe introduce some limit.
+        // TODO[RC]: Purge old statuses? - Remove the status from the manager upon putting the status into storage.
+        // TODO[RC]: Shall we store squeezed out variants as well?
```

Comment:
Hey @xgreenx 
I couldn't help but ponder about purging of the status manager. It is clear that we should purge statuses of transactions upon their insertion to storage, but this will cover the `Failed` and `Success` statuses only.

Now, regarding `SqueezedOut` - the question is: should we store them at all in the manager or should we just send them to the subscribers and forget. If the latter, the we don't need additional purging. This will also have a positive impact on DoS protection, since if we don't store `SqueezedOut` we don't need to care if people spam us with excessive numbers of transactions.

I think `Submitted` is not a problem since it'll eventually turn into other status.

cc: @AurelienFT